01_prepare_host.sh

```diff
@@ -6,24 +6,24 @@ source lib/logging.sh
 # shellcheck disable=SC1091
 source lib/common.sh
 
-if [[ $(id -u) == 0 ]]; then
+if [ "$(id -u)" -eq 0 ]; then
```

Comment:
When were running Bash for sure, `[[` should be used as its built-in (faster) and most compatible and feature rich. You seem to have changed all to `[`.